app/components/sidebars/main/channels_list/list/list.js

```diff
@@ -60,7 +68,8 @@ export default class List extends PureComponent {
         this.combinedActionsRef = React.createRef();
 
         this.state = {
-            sections: this.buildSections(props),
+            sections: this.props.showLegacySidebar ? this.buildSections(props) : [],
+            categorySections: this.props.showLegacySidebar ? [] : this.buildCategorySections(),
```

Comment:
Doesn't feel specially safe to use `this` in the constructor. Among other things, does "this.props" works correctly in this context? Shouldn't we be using the props in the function argument?

I would move `buildSections` and `buildCategorySections` outside of the class, to make sure we don't try to access things that may not be defined yet.